docs/plugin-write.md

```diff
@@ -0,0 +1,242 @@
+---
+id: plugin-write
+title: Writing Plugins
+---
+
+## Description
+
+Lets take a closer look at the minimal plugin instance we created in the [intro](plugin.md) part:
+
+```golang
+// Plugin is plugin instance
+type Plugin struct{}
+
+// Enable implements plugin.Plugin
+// Invoked just after initialization if plugin is already enabled.
+// Invoked every time the plugin is switched to enabled.
+func (c *Plugin) Enable() error {
+	return nil
+}
+
+// Disable implements plugin.Plugin
+// Invoked every time the plugin is switched to disabled.
+func (c *Plugin) Disable() error {
+	return nil
+}
+```
+
+Now only the base [`Plugin`](https://godoc.org/github.com/gotify/plugin-api#Plugin) interface has been implemented.
+To give it more functionalities, more interfaces can be implemented.
+
+## APIs
+
+### Intro
+
+The APIs are provided as interfaces and have generally two forms:
+
+The first form is called during plugin initialization.
+The methods are in the form of `^Register(.+)Handler$`, which plugins will implement to get a callback to the gotify engine to realize some functionalities.
+An example is [`Messenger`](https://godoc.org/github.com/gotify/plugin-api#Messenger), which is called with a callback that can be called at any time to send a message to the user.
+
+The second form is called after plugins are initialized (also after the initial `Enable()` call if the plugin is already enabled).
+They require the plugin to implements some methods which is called on user request to update information or provide feedbacks.
+An example is [`Displayer`](https://godoc.org/github.com/gotify/plugin-api#Displayer), which obtains an instruction generated by the plugin, which is then relayed to the user by gotify.
```

Comment:
Displayer#GetDisplay() is also called before the initial Enable(), so this is not. Lets just omit this "form" thingy completly. Because its explicitly defined on every plugin when it is called.